prometheus-to-sd/config/dynamic_source.go

```diff
@@ -18,8 +19,28 @@ const (
 	nameLabel    = "k8s-app"
 )
 
-// SourceConfigsFromDynamicSources takes pod specifications from the Kubernetes API and maps them to source configs.
-func SourceConfigsFromDynamicSources(gceConfig *GceConfig, sources []flags.Uri) ([]SourceConfig, error) {
+// ChangeType defines the possible types of events.
+type ChangeType string
+
+// Possible change types
+const (
+	Added    ChangeType = "ADDED"
+	Modified ChangeType = "MODIFIED"
+	Deleted  ChangeType = "DELETED"
+	Error    ChangeType = "ERROR"
+)
+
+// SourceConfigChange describes how one of the dynamic sources have changed.
+// Is generated when watch generates a new event.
+type SourceConfigChange struct {
+	ChangeType   ChangeType
+	SourceConfig SourceConfig
+	EventObject  string
+	Err          error
+}
+
+// GetChannelForDynamicSources takes pod specifications from the Kubernetes API and maps them to source configs.
+func GetChannelForDynamicSources(gceConfig *GceConfig, sources []flags.Uri) (<-chan SourceConfigChange, error) {
```

Comment:
I don't think adding Channel to names is good change in naming.

Reply:
done